app/controllers/articles_controller.rb

```diff
@@ -61,9 +61,18 @@ def create
     url = RequestHelper.url_from_param(params.dig(:article, :url))
     title, fetched_url = RequestHelper.extract_title_from_page(url)
 
-    new_article = Article.new(user: current_user, url: fetched_url, title: title)
+    # check if article already exists
+    article = current_user.articles.find_by_url(fetched_url)
 
-    if new_article.save
+    if article
+      article.update(url: fetched_url, title: title, updated_at: Time.now)
```

Comment:
We could only update the `updated_at` timestamp - the URL and title should remain the same anyway:

```suggestion
      article.update(updated_at: Time.now)
```